contrib/linux/bash-completion/dosbox

```diff
@@ -0,0 +1,183 @@
+# bash completion for dosbox                               -*- shell-script -*-
+# shellcheck shell=bash
+# vim: st=2:sw=2:et:ai:ft=bash
+### EDITING NOTES:
+# As this is a bash function, when adding variables or arrays it is important
+# to initialise with local. This prevents the var./array polluting the users
+# env. Please check output of:
+#    declare -x ; declare -a
+# after testing your new completion script
+#
+# Notable exception is COMPREPLY, this needs to persist for communication with
+# bash_completion
+####
+_dosbox()
+{
+  COMPREPLY=()
+  local cur prev words cword
+  _init_completion -s || return
+  local IFS=$'|\t\n'
+  local words_filter="${words[*]:1:$(( cword - 1 ))}"
+
+  ### Main options, note the \ line cont.
+  # using nameref
+  # shellcheck disable=SC2034
+  local -a \
+       _main_opts=(
+                    -h --help -fullscreen -startmapper -noautoexec -securemode
+                    -scaler -forcescaler -lang -machine -socket -exit
+                  )\
+     _repeat_opts=(
+                    -c -conf
+                  )\
```

Comment:
-userconf

Reply:
is -userconf fixed?

It also belongs in *_main_opts*
from what I understood it is 'oneshot' and replaces the base of the "layer cake" but it doesn't actually work, in essence it behaves in the same way as -conf does

Reply:
oh , ok In that case I miss understood it.

I that case I can add it, having it only consider those in *~/.config/dosbox-staging/*

Reply:
ok , I think I understand -userconf now

it forces the default config, so a partial .conf can be used

e.g.
dosbox-staging{-git}.conf has fullscreen set
foo.conf only contains [midi] section

- **dosbox -conf foo.conf**  *new* midi config is loaded, not fullscreen
- **dosbox -userconf -conf foo.conf** *new* midi config **and** fullscreen

only needed once, so added to _main_opts

Reply:
Confirmed the last two points are implemented in the completion script:

```
[08:05:17@server .bash_completion] touch test.conf
[08:05:20@server .bash_completion] dosbox -conf <TAB>
test.conf                dosbox.conf              dosbox-staging.conf      dosbox-staging-git.conf  
```

```
[08:05:20@server .bash_completion] dosbox -userconf -conf  <TAB>
test.conf                dosbox.conf              dosbox-staging.conf      dosbox-staging-git.conf  
```

Resolving.